modules/fbs/includes/fbs.availability.inc

```diff
@@ -32,106 +37,111 @@ function fbs_availability_items($provider_ids) {
  *   Holdings info.
  */
 function fbs_availability_holdings($provider_ids) {
-  $holdings = fbs_service()->Catalog->getHoldings(fbs_service()->agencyId, $provider_ids);
-  $result = array();
-  $tmp_issues = array();
-  foreach ($holdings as $item) {
-    $result[$item->recordId] = array(
-      'reservable' => $item->reservable,
-      // Gets properly set in the loop later.
-      'available' => FALSE,
-      'holdings' => array(),
-      'total_count' => 0,
-      // We don't know about this one.
-      'is_internet' => FALSE,
-      // And FBS can't tell us this.
-      'reserved_count' => 0,
-      'is_periodical' => FALSE,
-    );
-
-
-    $placement_parts = array('branch', 'department', 'location', 'sublocation');
-    foreach ($item->holdings as $item_holding) {
-      $result_holding = array(
-        'available_count' => 0,
+  try {
```

Comment:
Don't wrap everything with `try` blocks since it makes less reliable to catch other exceptions that might occur in the code below which is not related to fbs connectivity.
Only make sure that actual method calls are handled.

Seems that here u are filling holdings info, so in case of an exception just return an empty array. This way we get rid of additional code nesting level.